packages/components/src/font-size-picker/utils.js

```diff
@@ -18,15 +18,11 @@ const CUSTOM_FONT_SIZE_OPTION = {
  * Helper util to split a font size to its numeric value
  * and its `unit`, if exists.
  *
- * @param {string|number} size Font size.
+ * @param {string} size Font size.
  * @return {[number, string]} An array with the numeric value and the unit if exists.
  */
 export function splitValueAndUnitFromSize( size ) {
-	/**
-	 * The first matched result is ignored as it's the left
-	 * hand side of the capturing group in the regex.
-	 */
-	const [ , numericValue, unit ] = size.split( /(\d+)/ );
+	const [ numericValue, unit ] = size.match( /[\d\.]+|\D+/g );
```

Comment:
If I understand correctly the regex, if `size` doesn't contain any numeric value at all, `numericValue` won't actually be a numeric value, e.g.

```js
splitValueAndUnitFromSize('var(--some-css-variable)') // ['var(--some-css-variable)', undefined]
```

Reply:
You are correct, this was intended as a quick fix so that the hints would not "break" when decimal point values were used as they are in Twenty Twenty-Two. Ultimately a full overhaul of how these hints are generated is warranted. It does get very complicated once the theme creator uses a variety of different units. I am not sure we would need to account for everything as some onus needs to be on the theme developer. 🤔

Reply:
I haven't checked the code thoroughly, but I'm wondering why you changed the used `regex` function instead of just the regex.